externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -159,7 +159,9 @@ object SparkSQLEngine extends Logging {
       "20")
 
     val appName = s"kyuubi_${user}_spark_${Instant.now}"
+    val podNamePrefix = s"kyuubi-${user}-${Instant.now().toEpochMilli}"
```

Comment:
It's fine to me using `Instant.now().toEpochMilli` for unique. 

BTW we should move this into kubernetes case (Using `Utils.isOnK8s` help determine)

Reply:
The epochMilli LGTM, but still has confliction potential, and we have a better solution for that, see https://github.com/apache/incubator-kyuubi/issues/3345

Reply:
To be clear, either way is acceptable to me in this period.

Reply:
wait, why not use engineRefId here? it was passed by `spark.kyuubi.ha.engine.ref.id` to engine

Reply:
`spark.kyuubi.ha.engine.ref.id `is pretty long and adding it all up with an average username would result in the same error as reported here.

Reply:
from the documentation: 

> The prefix will be used to generate executor pod names in the form of podNamePrefix−exec−id, where the `id` is a positive int value, so the length of the `podNamePrefix` needs to be less than or equal to 47(= 63 - 10 - 6). 

and 

> You may use spark.kubernetes.executor.podNamePrefix to fully control the executor pod names. When this property is set, it’s highly recommended to make it unique across all jobs in the same namespace.

Reply:
Make sense, thanks.